File: lab5/haskel/diophantine.hs
de :: (Int, Int) -> (Int, Int, Int)
de (a, 0) = (1, 0, a)
de (a, b) =
    let (x', y', gcd) = de (b, (a `mod` b))
        (x, y) = (y', x' - y' * (a `div` b))
    in 
        (x, y, gcd)

File: lab5/prolog/diophantine.pl
de(A, 0, 1, 0, A).
de(A, B, X, Y, Z) :-
    B > 0,
    Tmp is A mod B,
    de(B, Tmp, X1, Y1, Z),
    X is Y1,
    Y is X1 - Y1 * (A // B).
